Mono.Addins/Mono.Addins.Description/AddinDescription.cs

```diff
@@ -1392,7 +1393,7 @@ void IBinaryXmlElement.Read (BinaryXmlReader reader)
 			nodeSets = (ExtensionNodeSetCollection) reader.ReadValue ("NodeSets", new ExtensionNodeSetCollection (this));
 			extensionPoints = (ExtensionPointCollection) reader.ReadValue ("ExtensionPoints", new ExtensionPointCollection (this));
 			conditionTypes = (ConditionTypeDescriptionCollection) reader.ReadValue ("ConditionTypes", new ConditionTypeDescriptionCollection (this));
-			fileInfo = (object[]) reader.ReadValue ("FilesInfo", null);
+			fileInfo = (AddinFileInfo[]) reader.ReadValue ("FilesInfo", null);
```

Comment:
This is not going to work. ReadValue can't infer the array type.

Reply:
Cool, thanks for the catch! I'll fix that

Reply:
But even if it's just object, `(AddinFileInfo[])(object[])` should just work